solidity-v1/dashboard/src/components/modal/threshold/WithdrawGrantedTokens.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useMemo, useState } from "react"
+import { withBaseModal } from "../withBaseModal"
+import { ModalBody, ModalFooter, ModalHeader } from "../Modal"
+import Button, { SubmitButton } from "../../Button"
+import { FormCheckboxBase } from "../../FormCheckbox"
+import { useFormik } from "formik"
+import useReleaseTokens from "../../../hooks/useReleaseTokens"
+import {
+  Accordion,
+  AccordionItem,
+  AccordionItemButton,
+  AccordionItemHeading,
+  AccordionItemPanel,
+} from "react-accessible-accordion"
+import { KEEP } from "../../../utils/token.utils"
+import TokenAmount from "../../TokenAmount"
+import { shortenAddress } from "../../../utils/general.utils"
+import moment from "moment"
+import { add } from "../../../utils/arithmetics.utils"
+import OnlyIf from "../../OnlyIf"
+
+export const WithdrawGrantedTokens = withBaseModal(({ grants, onClose }) => {
+  const [numberOfGrantsDisplayed, setNumberOfGrantsDisplayed] = useState(5)
+  const releaseTokens = useReleaseTokens()
+
+  const totalReadyToRelease = useMemo(() => {
+    return grants
+      .map((grant) => grant.readyToRelease)
+      .reduce((previous, current) => add(previous, current), 0)
+      .toString()
+  }, [grants])
+
+  const onWithdrawClick = (awaitingPromise) => {
+    for (const grantId of formik.values.grantIds) {
+      const selectedTokenGrant = grants.find((grant) => grant.id === grantId)
+      releaseTokens(selectedTokenGrant, awaitingPromise)
+    }
+    onClose()
+  }
+
+  const formik = useFormik({
+    enableReinitialize: true,
+    initialValues: {
+      grantIds: [],
+    },
+    onSubmit: (values) => {
+      alert(JSON.stringify(values, null, 2))
+    },
+  })
+
+  return (
+    <>
+      <ModalHeader>Withdraw</ModalHeader>
+      <ModalBody>
+        <h3 className={"mb-1"}>Withdraw granted tokens</h3>
+        <p className={"mb-2"}>
+          The following tokens are available to be released and withdrawn from
+          your token grant.
+        </p>
+        <div>
+          <div className="withdraw-granted-tokens__info-row">
+            <h4 className="withdraw-granted-tokens__info-row-title">
+              Withdraw:
+            </h4>
+            <span className="withdraw-granted-tokens__info-row-value">
+              total:{" "}
+              <TokenAmount
+                amount={totalReadyToRelease}
+                token={KEEP}
+                wrapperClassName="withdraw-granted-tokens__total-ready-to-release-amount"
+                amountClassName="text-grey-60"
+                symbolClassName="text-grey-60"
+              />
+            </span>
+          </div>
+          <form>
+            {grants
+              .slice(0, numberOfGrantsDisplayed)
+              .map((grant) => renderGrant(grant, grants.length, formik))}
+          </form>
+          <OnlyIf condition={grants.length > numberOfGrantsDisplayed}>
+            <div
```

Comment:
Can we use `Button` component. It will be correct in terms of semantic tags.

Reply:
Fixed in 13f3a49e47be6f4a0ed9be8868806d8a7ef3789f